DOC/CI: Prepare for v0.24 release

greglucas opened the pull request:
- Release notes
- Add Python 3.13 tests and builds

closes #2429

greglucas commented:
3.13 wheels are currently blocked by https://github.com/pyproj4/pyproj/issues/1425

rcomer commented:
pyproj 3.7 is out and CI is green :tada:

rcomer reviewed (approved):
This seems good to me.

greglucas commented:
@rcomer, I just pushed another two updates:
* Added the new LambertZoneII Projection to the list of projections
* Added v0.24 release notes to the what's new page

I'm not sure why the coastline cacheing test is failing on windows now...

rcomer commented on docs/source/reference/projections.rst:
```diff
@@ -341,6 +341,22 @@ OSGB
+
+    plt.figure(figsize=(3.2687, 3))
+    ax = plt.axes(projection=ccrs.LambertZoneII())
+    ax.coastlines(resolution='10m')
```
Actually maybe this one should add `BORDERS` rather than coastlines, so you get the outline of France?

rcomer replied:
Too late - never mind.

greglucas replied:
Sorry, just wanted to get this out the door so continued on! I agree with this though. I think there are a few that would be good to add borders to (if not all?).

rcomer commented:
The PyPI wheels only go up to py3.10 😕

QuLogic commented:
That's because the matrix generation pins `cibuildwheel` to an old version; I'm trying to see if I can fix it in #2450.
